pkg/controller/controller_ref_manager.go

```diff
@@ -349,7 +349,12 @@ func (m *ReplicaSetControllerRefManager) AdoptReplicaSet(rs *extensions.ReplicaS
 	// Note that ValidateOwnerReferences() will reject this patch if another
 	// OwnerReference exists with controller=true.
 	addControllerPatch := fmt.Sprintf(
-		`{"metadata":{"ownerReferences":[{"apiVersion":"%s","kind":"%s","name":"%s","uid":"%s","controller":true,"blockOwnerDeletion":true}],"uid":"%s"}}`,
+		`{"metadata":{
```

Comment:
@caesarxuchao the e2e deployment test is failing to create rs. It seems my patch is wrong. The relevant failure logs here https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/45764/pull-kubernetes-e2e-kops-aws/25580/